mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -446,6 +463,20 @@ private[ml] object DefaultParamsReader {
     val cls = Utils.classForName(metadata.className)
     cls.getMethod("read").invoke(null).asInstanceOf[MLReader[T]].load(path)
   }
+
+  def loadInitialModel[M <: Model[M]](instance: Params, path: String, sc: SparkContext): Unit = {
```

Comment:
Why are we checking the spark version here?

We can be more specific in the types:

````scala
  def loadInitialModel[M <: Model[M]](
      instance: HasInitialModel[M],
      path: String,
      sc: SparkContext): Unit = {
    implicit val format = DefaultFormats
    val metadata = DefaultParamsReader.loadMetadata(path, sc)

    val hasInitialModel = (metadata.metadata \ "initialModel").extract[Boolean]
    if (hasInitialModel) {
      val initialModelPath = new Path(path, "initialModel").toString
      val loadedInitialModel = loadParamsInstance[M](initialModelPath, sc)
      instance.set(instance.initialModel, loadedInitialModel)
    }
}
````